### ISSUE TITLE
Causal seedgen mode doesn't seem to work

### ISSUE THREAD

**Issue by Traehgniw** [clementgallet/SWD2Randomizer#8]
Tried causally genning a seed, got two seeds in a row with such things as grenade at Burster's aka speedrun logic.

**Comment by Traehgniw** [comment]
And now I just got Jet Engine at Burster's.


This would be great, if I had a way to exit the cave with it, akin to how the A Link to the Past randomizer has save&quit always return you to the start, or how the Ori randomizer allows you to alt-R back to the start at any point. It'd be like getting bash+grenade in an Ori seed. Suddenly like a gazillion things are in logic! Hooary! 

Instead, I am softlocked, because I did not get the singular pickup that would allow me to progress. And yes I tried save and quit, acts exactly like in vanilla Dig 2 and drops me back at the start of the cave without the pickup.

This is what the randomizer message was, by the way: 
"Selecting difficulty is Casual
Seed is 9158178
Extracting game data...
Query item flags: 
jackhammer, pressurebomb, steampack, fate.bloodquest, pressurebomb.launcher_triple, armor.damage_reduction, pickaxe.fire, hook.long_hook, pressurebomb.launcher, hook, fate.explosions, steampack.slayer, fate.xpx2, 
Randomize...........
Successfully found a correct case!
Compress into game data...
Done!"


and the seed is C9158178
